de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/model/AccessSqlParser.java

```diff
@@ -314,141 +328,168 @@ public String toString()
 		sb.append("|").append(getMainSqlIndex());
 		sb.append("]");
 		return sb.toString();
-	} //	toString
+	} // toString
 
 	/**
-	 * 	Get Table Info.
-	 * 	@param index record index
-	 *	@return table info
+	 * Get Table Info.
+	 * 
+	 * @param index record index
+	 * @return table info
 	 */
-	public TableInfo[] getTableInfo (int index)
+	public TableInfo[] getTableInfo(int index)
 	{
 		if (index < 0 || index > m_tableInfo.size())
 			return null;
-		TableInfo[] retValue = (TableInfo[])m_tableInfo.get(index);
+		TableInfo[] retValue = m_tableInfo.get(index);
 		return retValue;
-	}	//	getTableInfo
+	}	// getTableInfo
 
 	/**
-	 * 	Get Sql Statements
-	 * 	@param index record index
-	 *	@return index index of query
+	 * Get Sql Statements
+	 * 
+	 * @param index record index
+	 * @return index index of query
 	 */
-	public String getSqlStatement (int index)
+	public String getSqlStatement(int index)
 	{
 		if (index < 0 || index > m_sql.length)
 			return null;
 		return m_sql[index];
-	}	//	getSqlStatement
+	}	// getSqlStatement
 
 	/**
-	 * 	Get No of SQL Statements
-	 *	@return FROM clause count
+	 * Get No of SQL Statements
+	 * 
+	 * @return FROM clause count
 	 */
 	public int getNoSqlStatments()
 	{
 		if (m_sql == null)
 			return 0;
 		return m_sql.length;
-	}	//	getNoSqlStatments
+	}	// getNoSqlStatments
 
 	/**
-	 * 	Get index of main Statements
-	 *	@return index of main statement or -1 if not found
+	 * Get index of main Statements
+	 * 
+	 * @return index of main statement or -1 if not found
 	 */
 	public int getMainSqlIndex()
 	{
 		if (m_sql == null)
 			return -1;
 		else if (m_sql.length == 1)
 			return 0;
-		for (int i = m_sql.length-1; i >= 0; i--)
+		for (int i = m_sql.length - 1; i >= 0; i--)
 		{
 			if (m_sql[i].charAt(0) != '(')
 				return i;
 		}
 		return -1;
-	}	//	getMainSqlIndex
+	}	// getMainSqlIndex
 
 	/**
-	 * 	Get main sql Statement
-	 *	@return main statement
+	 * Get main sql Statement
+	 * 
+	 * @return main statement
 	 */
 	public String getMainSql()
 	{
 		if (m_sql == null)
 			return m_sqlOriginal;
-			
+
 		if (m_sql.length == 1)
 			return m_sql[0];
-		for (int i = m_sql.length-1; i >= 0; i--)
+		for (int i = m_sql.length - 1; i >= 0; i--)
 		{
 			if (m_sql[i].charAt(0) != '(')
 				return m_sql[i];
 		}
 		return "";
-	}	//	getMainSql
+	}	// getMainSql
 
 	/**
-	 * 	Table Info VO
+	 * Transform the key words WHERE, FROM and ON in lowercase.
+	 * TODO: Delete this logic when we get rid of the workaround in org.compiere.model.AccessSqlParser.getTableInfo(String)
+	 * 
+	 * @param whereClause
+	 * @return
+	 */
+	public String rewriteWhereClauseWithLowercaseKeyWords(final String whereClause)
+	{
+
+		StringBuilder whereClauseBuilder = new StringBuilder()
```

Comment:
there is no point in using StringBuilder in this case.

Reply:
done